### ISSUE TITLE
Consider doing some syntactical validation of Lua/WFL code

### ISSUE THREAD

**Issue by CelticMinstrel** [wesnoth/wesnoth#8385]
This feels like it might be asking a bit too much, but… I wonder if it would be reasonable for the schema to run at least a syntax check on any code included in keys that expect Lua or WFL code? Linting it would be even better, though that _really_ feels like it might not be worth it, but I think just a syntax check could be done by compiling it and checking if that returns an error…

There are a few other DSLs we have in WML, maybe those could be checked too (some of them might be checked already). For example, the CFG syntax for name generators or the progressive string syntax for animations.

**Comment by cooljeanius** [comment]
adding the "WML Tools" label because I think that'd be another place that it might be worth looking into doing this, besides the schema validator

**Comment by CelticMinstrel** [comment]
That's true – there's already a `luacheck` run in the CI, right? If that could somehow be adjusted to include Lua code embedded in WML, it would be quite helpful.

**Comment by CelticMinstrel** [comment]
I don't see anything in luacheck to make it synthesize line numbers, so I think the best way to handle that would be a Python script that reads a WML file, passes any Lua chunks to `luacheck` on `stdin` (using the `subprocess` module probably), and filter the output to adjust the error messages to refer to the original WML file instead of just the chunk.

**Comment by cooljeanius** [comment]
> I don't see anything in luacheck to make it synthesize line numbers (like C++'s `#line` directive), so I think the best way to handle that would be a Python script that reads a WML file, passes any Lua chunks to `luacheck` on `stdin` (using the `subprocess` module probably), and processes the output to adjust the error messages to refer to the original WML file instead of just the chunk.

A new Python script, or could it just be done as part of `wmllint`?

**Comment by CelticMinstrel** [comment]
Seems weird to insert `luacheck` into `wmllint`…

**Comment by cooljeanius** [comment]
> Seems weird to insert `luacheck` into `wmllint`…

I mean, it already runs an external spellchecking tool... also there seems to be some desire for something similar in #3817

**Comment by CelticMinstrel** [comment]
That issue doesn't seem to be about Lua though; and an external spellchecking tool is a bit different from making it call `luacheck`.
